modules/user_accounts/php/edit_user.class.inc

```diff
@@ -810,31 +810,39 @@ class Edit_User extends \NDB_Form
     }
 
     /**
-     * Query database on UserID to check that password has doesn't
-     * exist and account is still pending approval, to ensure the
-     * users with account activity can't be rejected by setting
-     * their status to pending.
+     * Query database on UserID to check that password hash does not
+     * exist and account is still pending approval.
      *
-     * @param string $id identifier of account to verify
+     * This ensures that users with account activity cannot be rejected
+     * if their approval status is, for some reason, set to pending.
      *
-     * @return boolean true is Password_hash is null and
-     * Pending_approval is Yes, false otherwsie
+     * @param string $userID of account to verify
+     *
+     * @return boolean true if 'Password_hash' is null and
+     *         'Pending_approval' is Yes. False, otherwise.
+     *
+     * @throws Loris exception if the userID is not found in the database.
      */
-    function _canRejectAccount($id)
+    static function canRejectAccount($userID)
     {
-        $DB     = \Database::singleton();
-        $fields =  $DB->pselect(
-            "SELECT Password_hash, Pending_approval
-                   FROM users
-                   WHERE UserID=:id",
-            array("id" => $id)
-        );
-        if (is_null($fields[0]['Password_hash'])
-            && $fields[0]['Pending_approval']=="Y"
-        ) {
-            return true;
+        $DB = \Database::singleton();
+        try{
+            $fields =  $DB->pselect(
+                "SELECT Password_hash, Pending_approval
+                           FROM users
+                           WHERE UserID=:id",
+                array("id" => $userID)
+            );
+            if (is_null($fields[0]['Password_hash'])
```

Comment:
empty is probably safer than is_null, since the password will never hash to "0", and I'm not sure if nulls ever get mangled to empty strings before getting here.